src/neo/Cryptography/MPT/MPTTrie.Delete.cs

```diff
@@ -0,0 +1,120 @@
+using Neo.IO;
+using System;
+using System.Collections.Generic;
+using static Neo.Helper;
+
+namespace Neo.Cryptography.MPT
+{
+    partial class MPTTrie<TKey, TValue>
+    {
+        public bool Delete(TKey key)
+        {
+            var path = ToNibbles(key.ToArray());
+            if (path.Length < 1) return false;
+            return TryDelete(ref root, path);
+        }
+
+        private bool TryDelete(ref MPTNode node, ReadOnlySpan<byte> path)
+        {
+            switch (node)
+            {
+                case LeafNode _:
+                    {
+                        if (path.Length < 1)
+                        {
+                            node = HashNode.EmptyNode;
+                            return true;
+                        }
+                        return false;
+                    }
+                case ExtensionNode extensionNode:
+                    {
+                        if (path.StartsWith(extensionNode.Key))
+                        {
+                            var result = TryDelete(ref extensionNode.Next, path[extensionNode.Key.Length..]);
+                            if (!result) return false;
+                            if (extensionNode.Next is HashNode hashNode && hashNode.IsEmpty)
+                            {
+                                node = extensionNode.Next;
+                                return true;
+                            }
+                            if (extensionNode.Next is ExtensionNode sn)
+                            {
+                                extensionNode.Key = Concat(extensionNode.Key, sn.Key);
+                                extensionNode.Next = sn.Next;
+                            }
+                            extensionNode.SetDirty();
+                            PutToStore(extensionNode);
+                            return true;
+                        }
+                        return false;
+                    }
+                case BranchNode branchNode:
+                    {
+                        bool result;
+                        if (path.Length < 1)
+                        {
+                            result = TryDelete(ref branchNode.Children[BranchNode.ChildCount - 1], path);
+                        }
+                        else
+                        {
+                            result = TryDelete(ref branchNode.Children[path[0]], path[1..]);
+                        }
+                        if (!result) return false;
+                        List<byte> childrenIndexes = new List<byte>();
```

Comment:
We can start the list with the `BranchNode.ChildCount` capacity.

Reply:
Done.